### PR TITLE
Proactive installation of an App using Graph API and sending proactive notifications

### PR DESCRIPTION
**Pull request by Trinetra-MSFT**
## Description

This sample app illustrates the proactive installation of app using Graph API and sending proactive notification to users from Group Chat or Channel.

Language Used: C#

### Proposed changes

- Integrate add app for user using graph API based on check and install commands
- Send proactive notification to user in personal scope 

### Testing
- Install the Proactive App Installation demo in a Team or GroupChat.
    image

- **Team Scope**: Run Check and install to pro-actively installs the App for all the users in team. After installation send 'Send message' command to send proactive message.
    image
- **Group Chat**:  Run Check and install to pro-actively installs the App for all the users in team. After installation send 'Send message' command to send proactive message.
   image

### CODE REVIEW

File: samples/graph-proactive-installation/csharp/README.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+2. Run the bot from a terminal or from Visual Studio:
+
+  - Launch Visual Studio
+  - File -> Open -> Project/Solution
+  - Navigate to `samples/graph-proactive-installation/csharp` folder
+  - Select `ProactiveBot.sln` file
+  - Press `F5` to run the project
+3. Run ngrok - point to port 3978
+   ```ngrok http -host-header=rewrite 3978``
+4. Create a new Bot by following steps mentioned in [Build a bot](https://docs.microsoft.com/en-us/microsoftteams/platform/bots/what-are-bots?view=msteams-client-js-latest#build--a-bot-for-teams-with-the-microsoft-bot-framework) documentation.
```
Comment (Wajeed-msft) [inline samples/graph-proactive-installation/csharp/README.md:49]: There is no mention of Graph API permissions in ReadMe file. Could you please add requried steps?

File: samples/graph-proactive-installation/csharp/proactive-Appinstallation-messages/Bots/ProactiveBot.cs [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+            string Userid = string.Empty;
+            string teamId = string.Empty;
+            string ChatId = string.Empty;
+            MicrosoftTenantId = turnContext.Activity.Conversation.TenantId;
+            var adaptiveAttachment = New_AdaptiveCardAttachment();
+            switch (turnContext.Activity.Conversation.ConversationType)
+            {
+                case "personal":
+                    Userid = turnContext.Activity.MembersAdded[0].AadObjectId;
+                    Tuple<bool, int> App_status = await Helper.AppinstallationforPersonal(Userid, MicrosoftTenantId, MicrosoftAppId, MicrosoftAppPassword, MicrosoftTeamAppid);
```
Comment (Wajeed-msft) [inline samples/graph-proactive-installation/csharp/proactive-Appinstallation-messages/Bots/ProactiveBot.cs:57]: This is getting triggered for the first time installation and failing due to 403 while calling Graph APIs. We need to request for Graph APIs first via login / application permission and then proceed with app installation.

**Review by Wajeed-msft — CHANGES_REQUESTED** [review]

File: samples/graph-proactive-installation/csharp/proactive-Appinstallation-messages/Bots/ProactiveBot.cs (hunk 2 above)
Comment (Trinetra-MSFT) [reply samples/graph-proactive-installation/csharp/proactive-Appinstallation-messages/Bots/ProactiveBot.cs]: Need to confirm this, if we are going to add delegated auth flow to get the teamAppId generated from catalog. As far as i know application permissions is not  supported for App Catalog, if we need to add sign in flow when we can ask user to sign ?

File: samples/graph-proactive-installation/csharp/README.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  - Press `F5` to run the project
+3. Run ngrok - point to port 3978
+   ```bash
+   ngrok http -host-header=rewrite 3978
+4. Go to appsettings.json and add ```MicrosoftAppId``` ,  ```MicrosoftAppPassword```, ```AppCatalogTeamAppId``` information.
+5. Run your app, either from Visual Studio with ```F5``` or using ```dotnet run``` in the appropriate folder.
+6. Update the manifest.json file with ```Microsoft-App-ID```, ```Base Url```   value.
+
+
+### Descriptions MS TeamsApp resource type
```
Comment (Wajeed-msft) [inline samples/graph-proactive-installation/csharp/README.md:35]: What does this heading means?

File: samples/graph-proactive-installation/csharp/README.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  - Launch Visual Studio
+  - File -> Open -> Project/Solution  
+  - Navigate to `samples/graph-proactive-installation/csharp` folder
+  - Select `ProactiveBot.sln` file
+  - Press `F5` to run the project
+3. Run ngrok - point to port 3978
+   ```bash
+   ngrok http -host-header=rewrite 3978
+4. Go to appsettings.json and add ```MicrosoftAppId``` ,  ```MicrosoftAppPassword```, ```AppCatalogTeamAppId``` information.
+5. Run your app, either from Visual Studio with ```F5``` or using ```dotnet run``` in the appropriate folder.
```
Comment (Wajeed-msft) [inline samples/graph-proactive-installation/csharp/README.md:31]: Could you pleas eadd details on how to fetch AppCatalogTeamAppId?

File: samples/graph-proactive-installation/csharp/README.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+- List of Apps in Groupchat
+- Add App to Groupchat
+- Add App to the Personal scope of Groupchat members
+
+
+1. Graph API Authentication  using Application Permissions
+    Deploy this app to Azure and give API permissions as Application.For more information (https://docs.microsoft.com/en-us/graph/auth-v2-service)
+2. List of apps in team
+    Retrieve a list of apps installed in the specified team [To get the details we need application Permission].
+    For more information (https://docs.microsoft.com/en-us/graph/auth-v2-service)
```
Comment (Wajeed-msft) [inline samples/graph-proactive-installation/csharp/README.md:49]: Intenstion of this section is not clear. Could you please check and appropriate changes?

**Review by Wajeed-msft — CHANGES_REQUESTED** [review]

**Review by Wajeed-msft — APPROVED** [review]

**Review by Sudhir-MSFT — APPROVED** [review]

**Pull request merged**